### ISSUE TITLE
User should not be allowed to select current class as logical definition superclass

### ISSUE THREAD

**Issue by soimugeoWB** [who-icatx/icatx-project#127]
As a user i should be able to select a logical definition superclass.
As a user i am not allowed to select the current class as logical definition superclass (e.g. cholera to have cholera as logical definition superclass).

If this scenario happens then we create a cycle in the hierarchy. 
![Image](https://github.com/user-attachments/assets/d84442c7-f0eb-476e-a7b7-26031bb2cb95)

Also the edit parents functionality to delete this cycle is not working.

**Comment by tudorache** [comment]
Yes, this is a bug. The UI should prevent using the selected class as the logical definition superclass. The change parents feature should be able to delete the class as its own superclass, but I guess this is a different bug.
